pkg/lakom/resolvetag/admission.go

```diff
@@ -140,63 +148,169 @@ func (h *handler) Handle(ctx context.Context, request admission.Request) admissi
+			if err != nil {
+				return nil, err
+			}
+			extension.Spec.Deployment.AdmissionDeployment.RuntimeCluster.Helm.OCIRepository.Ref = &resolved
+		}
+
+		if extension.Spec.Deployment != nil &&
+			extension.Spec.Deployment.AdmissionDeployment != nil &&
+			extension.Spec.Deployment.AdmissionDeployment.VirtualCluster != nil &&
+			extension.Spec.Deployment.AdmissionDeployment.VirtualCluster.Helm != nil &&
+			extension.Spec.Deployment.AdmissionDeployment.VirtualCluster.Helm.OCIRepository != nil {
+			resolved, err := h.handleContainer(ctx, extension.Spec.Deployment.AdmissionDeployment.VirtualCluster.Helm.OCIRepository.GetURL(), kcr, logger)
+			if err != nil {
+				return nil, err
+			}
+			extension.Spec.Deployment.AdmissionDeployment.VirtualCluster.Helm.OCIRepository.Ref = &resolved
 		}
+
+		if extension.Spec.Deployment != nil &&
+			extension.Spec.Deployment.ExtensionDeployment != nil &&
+			extension.Spec.Deployment.ExtensionDeployment.Helm != nil &&
+			extension.Spec.Deployment.ExtensionDeployment.Helm.OCIRepository != nil {
+			resolved, err := h.handleContainer(ctx, extension.Spec.Deployment.ExtensionDeployment.Helm.OCIRepository.GetURL(), kcr, logger)
+			if err != nil {
+				return nil, err
+			}
+			extension.Spec.Deployment.ExtensionDeployment.Helm.OCIRepository.Ref = &resolved
+		}
+
+		bytes, err = json.Marshal(extension)
+		if err != nil {
+			return nil, err
+		}
+	case podGVK:
+		pod := corev1.Pod{}
+		err := h.decoder.Decode(request, &pod)
+		if err != nil {
+			return nil, err
+		}
+
+		kcr = utils.NewLazyKeyChainReaderFromPod(ctx, h.reader, &pod, h.useOnlyImagePullSecrets)
+
+		for idx, ic := range pod.Spec.InitContainers {
+			resolved, err := h.handleContainer(ctx, ic.Image, kcr, logger)
+			if err != nil {
+				return nil, err
+			}
+
+			pod.Spec.InitContainers[idx].Image = resolved
+		}
+		for idx, c := range pod.Spec.Containers {
+			resolved, err := h.handleContainer(ctx, c.Image, kcr, logger)
+			if err != nil {
+				return nil, err
+			}
+
+			pod.Spec.Containers[idx].Image = resolved
+		}
+		for idx, ec := range pod.Spec.EphemeralContainers {
+			resolved, err := h.handleContainer(ctx, ec.Image, kcr, logger)
+			if err != nil {
+				return nil, err
+			}
+
+			pod.Spec.EphemeralContainers[idx].Image = resolved
+		}
+		bytes, err = json.Marshal(pod)
+		if err != nil {
+			return nil, err
+		}
+	default:
+		return nil, fmt.Errorf("unknown kind %v", request.Kind.Kind)
+	}
+
+	return bytes, nil
+}
+
+// getURL returns the fully-qualified OCIRepository URL of the image.
+func getURL(img *seedmanagement.Image) string {
+	ref := *img.Repository
+
+	if img.Tag != nil {
+		ref = ref + ":" + *img.Tag
 	}
 
-	return nil
+	return strings.TrimPrefix(ref, "oci://")
 }
 
 func (h *handler) handleContainer(ctx context.Context, image string, kcr utils.KeyChainReader, logger logr.Logger) (string, error) {
```

Comment:
Now it makes sense to rename this method to `handleImage` or `resolveImage` because it is no longer working on container images only.